### DISCUSSION TITLE
RPi Pico as SPI slave

### DISCUSSION THREAD

**Discussion by Myungjin-Jung** [micropython/micropython#15352] [RP2040 / Pico]
I am going to build multiple RPi Picos as SPI slaves connected to one RPi pico as a master using micropython.
And, I cannot find any examples or documents on making RPi Pico an SPI slave.
I have checked the sck pins, and it looks like the clock from the master is somewhat attenuated and was mixed with the slave's clock signal.
Is there any help I can get?
Thank you.

**Comment by robert-hh** [comment]
The RP2040 port does not support SPI slave mode. The only ports that do are STM32 and CC3200.

**Reply by GitHubsSilverBullet** [reply]
You could set the SPI registers (datasheet) directly, using mem32, @viper or @asm_thumb.
If you combine that with the DMA capabilities, you're ready to go.
